web/packages/teleterm/src/ui/appContext.ts

```diff
@@ -174,29 +172,35 @@ export default class AppContext implements IAppContext {
   }
 
   private setUpTshdEventSubscriptions() {
-    this.subscribeToTshdEvent('relogin', ({ request, onCancelled }) => {
-      // The handler for the relogin event should return only after the relogin procedure finishes.
-      return this.reloginService.relogin(
-        request as ReloginRequest,
-        onCancelled
-      );
-    });
+    this.setupTshdEventContextBridgeService({
+      relogin: async ({ request, onCancelled }) => {
+        await this.reloginService.relogin(
+          request as ReloginRequest,
+          onCancelled
+        );
+        return {};
+      },
```

Comment:
`request` and the returned response are fully type-checked according to the relevant gRPC handler. You can test this out by writing this:

```
      promptMFA: async ({request}) => {
        request.foo

        return { totpCode: false }
      },
```

TypeScript will complain that `foo` doesn't exist on `request` and that the type of `totpCode` is wrong (it should be a string).

Unfortunately, or maybe fortunately, each function here has to have an explicit return. This is because technically each gRPC handler has its own response message. So even if we end up always returning an empty `ReloginResponse`, this place has to return an empty object just to type check (so that return type is `object` and not `void`). I'm fine with that, especially since adjusting for this would require even more complex types for `TshdEventContextBridgeService`.